### ISSUE TITLE
Mining page, and one charting library instead of two abandoned ones

### ISSUE THREAD

**Issue by NiKrause** [Le-Space/explorer#16]
A mining page like the one on [explorer.doichain.org/mining](https://explorer.doichain.org/mining/) — and, while we are in there, one charting library instead of the two abandoned ones we carry today.

## What the reference page shows

| | |
|---|---|
| **Tiles** | last block reward · 24 h average reward · hashrate · difficulty |
| **Charts** | hashrate & difficulty over 7 days (two axes) · reward per block over 24 h · top miners over 24 h |
| **Tables** | top miners · recent blocks (time, height, hash, winner, reward, fees) |

## The part that is not obvious: we have no block history

This is the whole of the work, and it is worth stating before anyone reaches for a charting library.

**There is no `blocks` collection.** `models/` holds `tx`, `address`, `addresstx`, `richlist`, `stats`, `markets`, `peers`, `heavy` — and no blocks. Block pages are rendered by asking the daemon over RPC at request time, which is why they work at all.

**`stats` keeps only current values**, and the two fields this page needs most are commented out in the schema:

```js
//difficulty: { type: Object, default: {} },
//hashrate: { type: String, default: 'N/A' },
```

They are fetched live through `lib/explorer.js` on every page load. Nothing anywhere records what the difficulty *was*.

So "last 7 days" has nothing to draw from. At 600 s target spacing that is roughly 1 000 blocks, and asking the node for them on each page view is not an option.

## Suggested implementation, in stages

**1 — record blocks as they are synced.** A `blocks` collection written by `scripts/sync.js` alongside the transactions it already walks:

```
height, hash, time, difficulty, size, txcount,
reward (coinbase out), fees, winner (coinbase output address)
```

Plus a one-off backfill for the last N days. This stage is the bulk of it and is independently useful — a `/blocks` list, an "average block time" figure and anything else time-based all become possible.

**2 — aggregate endpoints**, so the browser fetches small JSON rather than thousands of rows:

- `/ext/mining/series?days=7` → buckets of `{ t, difficulty, hashrate }`
- `/ext/mining/rewards?hours=24` → `{ t, reward, fees }` per block
- `/ext/mining/miners?hours=24` → `{ winner, blocks, reward }`

**3 — the page itself**, `/mining`, with the tiles, three charts and two tables, in both languages like the rest.

**4 — one charting library**, replacing what is there now.

## Which charting library

Today the explorer carries **two, both abandoned**:

- **jqPlot 1.9pre** in `public/vendor/jqplot/` — last release 2013 — draws the wealth pie in `views/richlist.pug`
- **Chart.js in its 1.x API** in `public/javascripts/chart.js` — `new Chart(ctx).Doughnut(data)` has not been the API since 2016 — draws both charts in `views/reward.pug`

Plus **DataTables from a CDN** for the tables.

**Recommendation: Chart.js 4, self-hosted.**

- One library covers all three shapes this page needs — time series with two axes, bars, doughnut — and the wealth pie that jqPlot draws today.
- **MIT**, which sits cleanly beside the BSD-3 this project is under.
- **Self-hosted, and that matters here.** This page currently pulls from seven third-party hosts (`ajax.googleapis.com`, `cdn.datatables.net`, `cdnjs.cloudflare.com`, `fonts.googleapis.com`, `fonts.gstatic.com`, `maxcdn.bootstrapcdn.com`, `stackpath.bootstrapcdn.com`), each of which sees every visitor's IP address. Adding an eighth would be the wrong direction; a vendored file adds none.
- The shortest migration from what is already there: `reward.pug` moves from the 1.x API to 4.x, `richlist.pug` loses jqPlot, and the vendor folder gets smaller.
- Themable from the CSS custom properties the brand layer already defines, so the charts inherit the palette instead of repeating it.

**Considered and not chosen:** [ECharts](https://echarts.apache.org/) (Apache-2.0) is more capable — built-in zoom, comfortable with very large series — at three to five times the size; worth revisiting if we ever want candlesticks or a full-chain view. **amCharts**, which the reference page uses, needs its licence terms checked before adoption: the free tier is not unconditional.

## Two things to get right in the data

**Hashrate is derived, not measured.** There is no such number on a chain; it is inferred from difficulty and the observed spacing. The label should say so, and the chart should use the same formula as `getnetworkhashps` rather than inventing a second one.

**Miner attribution on a merge-mined chain.** The "winner" is the coinbase output address, which identifies a payout destination and not necessarily an operator — one operator can use many, and a pool pays many. The distribution can also be concentrated enough that a pie chart is a single wedge, so the top-miners view needs to stay readable with very few distinct entries. Worth labelling as "blocks by coinbase address" rather than "miners", which is what it actually is.
